src/cli_agent_orchestrator/services/settings_service.py

```
@@ -95,6 +96,54 @@ def is_memory_enabled() -> bool:
+
+def get_compile_timeout_s() -> float:
+    """Return the wall-clock timeout (seconds) for the wiki compile call.
+
+    Generous by default: compilation drives a coding-agent CLI that can
+    cold-start in tens of seconds, and it runs in the background so the
+    timeout never blocks store().
+    """
+    try:
+        value = get_memory_settings().get("compile_timeout_s", 120.0)
+        return float(value)
+    except Exception as e:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)